File: MigratoryBirds.js
function migratoryBirds(arr) {
    let birdsMap = {};
    for (let i = 0; i < arr.length; i++) {
        let birdType = arr[i];
        let count = birdsMap[birdType];
        if (count === undefined) {
            count = 0;
        }
        birdsMap[birdType] = count + 1;
    }

    let lowestType = Number.MAX_SAFE_INTEGER;
    let highestCount = 0;
    for (let birdType in birdsMap) {
        if (birdsMap.hasOwnProperty(birdType)) {
            let count = birdsMap[birdType];
            if (count > highestCount || (count === highestCount && birdType < lowestType)) {
                highestCount = count;
                lowestType = birdType;
            }
        }
    }

    return lowestType;
}

File: MigratoryBirds.py
def migratory_birds(arr):
    birds_map = {}
    for bird_type in arr:
        count = birds_map.get(bird_type, 0)
        birds_map[bird_type] = count + 1

    highest_type = None
    highest_count = 0
    for bird_type, count in birds_map.items():
        if count >= highest_count:
            highest_count = count
            highest_type = bird_type

    return highest_type